lib/cmdlib/cluster/__init__.py

```diff
@@ -1367,6 +1367,12 @@ def CheckPrereq(self):
             else:
               self.new_os_hvp[os_name][hv_name].update(hv_dict)
 
+      # Cleanup any OS that has an empty hypervisor parameter list, as we don't
+      # need them in the cluster config anymore.
+      for os_name, hvs in self.new_os_hvp.items():
+        if not hvs:
+          self.new_os_hvp.pop(os_name, None)
+
```

Comment:
Does this mean in order to remove a OS, we need to remove associated hv params? Could you please add a note in the respective man page?